src/nnet3/nnet-am-decodable-simple.cc

```diff
@@ -217,7 +226,7 @@ void NnetDecodableBase::DoNnetComputation(
   int32 time_offset = (shift_time ? -output_t_start : 0);
 
   // First add the regular features-- named "input".
-  request.inputs.reserve(2);
+  request.inputs.reserve(2 + recurrent_output_names_.size());
```

Comment:
@pegahgh and @vimalmanohar could you confirm that these changes are compatible with the changes you are making to egs format for supporting new cost functions.